### PR TITLE
allow create a subscriber with dead-lettering & exponential retries

### PR DESCRIPTION
**Pull request by aymaneMx**
### :tophat: What?

allow create a subscriber with dead-lettering & exponential retries

### :thinking: Why?

we need a way to configure those 2 parameters programmatically 

### :link: Related issue

Fix https://github.com/mercadona/rele/issues/222

### CODE REVIEW

File: rele/worker.py [hunk 1]
```diff
@@ -144,6 +152,13 @@ def create_and_run(subs, config):
         config.credentials,
         config.ack_deadline,
         config.threads_per_subscription,
+        config.retry_policy,
+        {
+            "dead_letter_topic_id": config.dead_letter_topic_id,
+            "max_delivery_attempts": config.max_delivery_attempts,
+        }
+        if config.dead_letter_topic_id
+        else None,
```
Comment (andrewgy8) [inline rele/worker.py:161]:
```suggestion
       config.dead_letter
```
Id suggest that the config should be handling the deadletter, and know all about how to set it up.

File: docs/settings.rst [hunk 2]
```diff
@@ -160,3 +166,33 @@ library behavior, please set this value to 10.
...
 Boolean function that applies a global filter on all subscriptions.
 For more information, please see `Filtering Messages section <https://mercadonarele.readthedocs.io/en/latest/guides/filters.html#global-filter>`.
+
+
+``DEFAULT_RETRY_POLICY``
+----------------------------
+
+**Optional**
+
+Dictionary with two keys: `minimum_backoff` and `maximum_backoff`, that specifies how Pub/Sub retries message delivery for all the subscriptions.
```
Comment (jonasae) [inline docs/settings.rst:176]:
```suggestion
Dictionary with two keys: `minimum_backoff` and `maximum_backoff`, that specifies how Pub/Sub retries message delivery for all the subscriptions in seconds.
```

File: docs/settings.rst [hunk 3]
```diff
@@ -160,3 +166,33 @@ library behavior, please set this value to 10.
...
+
+    If not set, the default retry policy is applied. This generally implies that messages will be retried as soon as possible for healthy subscribers. RetryPolicy will be triggered on NACKs or acknowledgement deadline exceeded events for a given message.
```
Comment (jonasae) [inline docs/settings.rst:178]:
```suggestion
    If not set, the default retry policy is applied, meaning a minimum backoff of 10 seconds and a maximum backoff of 60 seconds. 
    This generally implies that messages will be retried as soon as possible for healthy subscribers. RetryPolicy will be triggered on NACKs or acknowledgement deadline exceeded events for a given message.
```

**Comment by jonasae** [comment]
We are splitting and making some changes from this PR.
Thanks for collaborating ;-)

**Pull request closed without merging**